### ISSUE TITLE
Media library: cannot add/replace images in narrow viewport widths

### ISSUE THREAD

**Issue by ramonjd** [WordPress/gutenberg#60152]
### Description

When adding an item from the Media Library, the Select button is unusable in narrow viewport widths.

An overlay covers it, effectively disabling selecting an image. 

The button, however, is not disabled via any attribute, therefore it probably constitutes an accessibility problem as well.

This is probably a Core issue, but reporting here in case the block editor needs to provide the exception to the rule (if there is one).

The issue, as far as I can see, was caused by a [recent Core commit](https://core.trac.wordpress.org/changeset/57605), in particular, the `.media-bg-overlay` CSS rule and associated media query:

https://github.com/WordPress/wordpress-develop/commit/1a4692f54d1863386549dfc0a3ffd7a40076c1cc#diff-c657d8edf83f580c59e61e205f967175fbf6e687c188e87e98221b47286c7074R2832

Related trac ticket: https://core.trac.wordpress.org/ticket/33049

### Step-by-step reproduction instructions

1. Insert an Image block
2. Click on "Media library" to insert an image
3. Narrow the viewport width to < 782px
4. Observe that the "Select" button is disabled by an overlay

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/6458278/ea27ba9b-8282-4790-b9d1-4ec59e219533

### Environment info

WordPress 6.6-alpha-57778-src running Twenty Twenty-Four theme.

With Gutenberg trunk activated and disabled

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ramonjd** [comment]
This Core patch should fix it: https://github.com/WordPress/wordpress-develop/pull/6313

**Comment by ramonjd** [comment]
Closing in favour of https://core.trac.wordpress.org/ticket/33049#comment:15 and https://github.com/WordPress/wordpress-develop/pull/6313
